### DISCUSSION TITLE
Can we use/install mods?

### DISCUSSION THREAD

**Discussion by evolved311** [lloesche/valheim-server-docker#191] [Q&A] [answered]
Sorry, I'm a real novice with all this, but I do have lloesche's valheim-server-docker up and running perfectly on my Synology NAS.  I was just wondering if it was possible to install mods onto the server (such as "Running Skill Give More Carry Weight" available on Nexus Mods)?  From the instructions it looks like it also requires the installation of "BepInEx".

If it's possible to install these types of mods, can someone give me instructions on how to do it with a Synology NAS?  Thank you!

**Comment by lloesche** [comment, marked as answer]
@evolved311 the only mod this container officially supports right now is ValheimPlus. See https://github.com/lloesche/valheim-server-docker#valheimplus for details.

ValheimPlus uses BepInEx so you could add more mods to that installation but then you're pretty much on your own as far as support goes. Also any Valheim server or ValheimPlus updates would reset those mods again.

For advanced users there are PRE/POST hooks (https://github.com/lloesche/valheim-server-docker#event-hooks) that allow you to run all kinds of commands when certain things in the container happen (like updates). So you could automate the installation of additional mods, but again, more of an advanced user / developer feature.

**Reply by evolved311** [reply]
Thanks so much for the information, Lukas!  I was able to get ValheimPlus up and running, so that should be enough for now. :)

For anyone else who might be having trouble, the only way I could get it to create the config file (on Synology NAS using Docker) was to CLEAR the container, then download it in the Registry again, then start it back up.  Without clearing/redownloading it would never create the config file.

**Reply by lloesche** [reply]
I think that was likely just the bug we fixed like 2h ago. The new ValheimPlus default config included some new entry that the config parser did not like. So downloading the latest version of the container is what fixed the issue. Clearing the data should not be neccessary.

**Reply by evolved311** [reply]
Ahhh!  Got it.  Thanks again!

**Comment by MAL0W** [comment]
> Sorry, I'm a real novice with all this, but I do have lloesche's valheim-server-docker up and running perfectly on my Synology NAS. I was just wondering if it was possible to install mods onto the server (such as "Running Skill Give More Carry Weight" available on Nexus Mods)? From the instructions it looks like it also requires the installation of "BepInEx".
> 
> If it's possible to install these types of mods, can someone give me instructions on how to do it with a Synology NAS? Thank you!

Could you tell me how you installed PLUS MOD and BepInEx on your Synology? I also have it with Synology.

THANKS!!

**Reply by lloesche** [reply]
@MAL0W when you say PLUS MOD do you mean ValheimPlus? If so the process is described here: https://github.com/lloesche/valheim-server-docker#valheimplus

**Reply by MAL0W** [reply]
Thanks Lukas, let's see if I can make it work since my experience with docker is 0,2

Thanks again for your work. Whenever you want, I invite you to a beer / cafe by paypal :)
